### PR TITLE
Chore/4624 test blotter

### PR DESCRIPTION
**Pull request by Clark-Adaptive**

### CODE REVIEW

**Review by joeg-adaptive — DISMISSED** [review]

**Review by joeg-adaptive — DISMISSED** [review]

File: src/client/e2e/blotter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    await tile.click()
+
+    const tradeId = await tilePage
+      .locator("[data-testid='trade-id']")
+      .innerText()
+    const newRow = await blotterPage.getByTestId(`trades-grid-row-${tradeId}`)
+    await expect(newRow).toHaveCSS("animation", /1s ease-in-out/)
+  })
+  test("when user clicks on the header of any column, it should sort it (depending on number of clicks, can be ascending or descending)", async () => {
+    const firstTradeID = await blotterPage
```
Comment (iprado78) [inline src/client/e2e/blotter.spec.ts:61]: just for a little more clarity, call it `firstRowTradeID`

Comment (iprado78) [reply src/client/e2e/blotter.spec.ts:61]:
```
await blotterPage.locator('[role="grid"] > tr')
  .nth(0)
  .locator('td')
  .nth(getTradeIdColIndex())
  .textContent()
```

This makes explicit that you are first selecting the first row, then the trade id cell in it.

File: src/client/e2e/blotter.spec.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      .locator('[role="grid"] > tr > td')
+      .nth(getTradeColIndex())
+      .textContent()
+
+    const sortTradeID = await blotterPage.getByText("Trade ID", { exact: true })
```
Comment (iprado78) [inline src/client/e2e/blotter.spec.ts:66]: again, clearer if it's `tradeIDColHeader` - let's me visualize it like a user

File: src/client/e2e/blotter.spec.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      .locator('[role="grid"] > tr > td')
+      .nth(getTradeColIndex())
+      .textContent()
+
+    expect(afterClick1).toBe(firstTradeID)
+
+    // filter 2nd time (ascending)
+    await sortTradeID.click()
+    const afterClick2 = await blotterPage
+      .locator('[role="grid"] > tr > td')
```
Comment (iprado78) [inline src/client/e2e/blotter.spec.ts:80]: pattern/function

File: src/client/e2e/blotter.spec.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    })
+    await tradeIDColHeader.hover()
+    const filterButton = await blotterPage.locator(
+      '[aria-label*="Open Trade ID"]',
+    )
+    await filterButton.click()
+    const searchInput = await blotterPage.locator(
+      '[aria-label*="Primary filter"]',
+    )
+    const searchedRow = "1"
```
Comment (iprado78) [inline src/client/e2e/blotter.spec.ts:98]: `const tradeIDToSearch = ...  // Name change for clarity`

File: src/client/e2e/blotter.spec.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      csvRows = "error"
+    }
+    expect(typeof csvRows).not.toBe("string")
+    // get index 1 to account for header row
+    const firstRowFields = csvRows[1].split(",")
+    const tradeField = firstRowFields[0]
+    expect(tradeField).toBe(firstTradeID)
+  })
+  test("when user filters blotter and then clicks export button on blotter, the csv file should be filtered", async () => {
+    const tradeIDColHeader = await blotterPage.getByText("Trade ID", {
```
Comment (iprado78) [inline src/client/e2e/blotter.spec.ts:140]: good name!

File: src/client/e2e/blotter.spec.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    })
+    await tradeIDColHeader.hover()
+    const filterButton = await blotterPage.locator(
+      '[aria-label*="Open Trade ID"]',
+    )
+    await filterButton.click()
+    const searchInput = await blotterPage.locator(
+      '[aria-label*="Primary filter"]',
+    )
+    const searchedRow = "2"
```
Comment (iprado78) [inline src/client/e2e/blotter.spec.ts:151]: as above: `const tradeIDToSearch``

File: src/client/e2e/blotter.spec.ts [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    const downloadButton = await blotterPage.locator(
+      '[aria-label="Export to CSV"]',
+    )
+    await downloadButton.click()
+    const download = await downloadPromise
+    await download.saveAs("e2e/test-data/filtered-data.csv")
+    //read the downloaded file, use readFileSync because it is synchronous
+    let filteredData: Array<string> | string
+    try {
+      const data = fs
```
Comment (iprado78) [inline src/client/e2e/blotter.spec.ts:164]: don't need intermediate variable, `filteredCSVRows = `

**Review by iprado78 — CHANGES_REQUESTED** [review]

**Review by iprado78 — APPROVED** [review]

**Pull request merged**